details.js

```diff
@@ -0,0 +1,24 @@
+function getCountry() {
+  document.addEventListener("DOMContentLoaded", () => {
+    const countryDetails = JSON.parse(localStorage.getItem("countryDetails"));
```

Comment:
Couple of things regarding this function : 

1) Make sure that this function is only doing 1 thing, right now the function really overloaded with different things to do
2) You can extract the localStorage logic into service and call the service methods from there
3) Do not repeat `document.querySelector(".country-flag img")`, save this inside a variable, and one more thing that you should do is to style your element via css classes and not js
4) Do not use `innerHTML`, it is a dangerous method